osbs/core.py

```diff
@@ -437,10 +438,18 @@ def stream_logs(self, build_id):
                 for line in decoded_json(response.iter_lines()):
                     last_activity = time.time()
                     yield line
-            except (ChunkedEncodingError,
-                    ConnectionError,
-                    httplib.IncompleteRead):
-                pass
+            # NOTE1: If self._get causes ChunkedEncodingError, ConnectionError,
+            # or IncompleteRead to be raised, they'll be wrapped in
+            # OsbsNetworkException or OsbsException
+            # NOTE2: If decode_json or iter_lines causes ChunkedEncodingError, ConnectionError,
+            # or IncompleteRead to be raised, it'll simply be silenced.
+            # NOTE3: An error may occur in iter_lines during initial contact
+            # with server - prior to streaming data. In this case, exception will be
+            # wrapped in OsbsException or OsbsNetworkException, inspect cause
+            # to detect ConnectionError.
```

Comment:
As you point out in note 2,  `iter_lines` silences ConnectionError, so the only thing that can raise OsbsException is `check_response`. But it can and will.